nexus/src/external_api/http_entrypoints.rs

```diff
@@ -6442,14 +6442,30 @@ impl NexusExternalApi for NexusExternalApiImpl {
                 crate::context::op_context_for_external_api(&rqctx).await?;
             let params = body.into_inner();
             let system_version = params.system_version;
+
+            let current_target_release =
```

Comment:
Cool.  I like the test, too.

I wonder if this condition needs to be checked transactionally with the update, though?  I'm thinking of the time-of-check-to-time-of-use race where say the system is at version 14, the user is trying to update to 15, but someone else comes in and sets it to 16 in the meantime, then we overwrite that with 15.  But maybe this is prevented because at L6469 below, we construct the next target release with a generation number of 1 + whatever we just saw (that was version 14), so we'll necessarily fail the database update if someone else comes in in between?  If that's the case, could we add a comment here explaining why this race isn't a problem?

Reply:
That is the case; documented in c2233b941.